### PR TITLE
Address DH feedback on pos fee estimation v2

### PR DESCRIPTION
**Pull request by lazynina**
- Adds separate params for `numPastBlocks` and `numMempoolBlocks` so we can tune estimator differently for past blocks estimator and mempool estimator.
- Removes "in-between" static fee case
- Adds separate fee estimation params for `mempoolCongestionFactorBasisPoints`, `mempoolPriorityPercentileBasisPoints`, `pastBlocksCongestionFactorBasisPoints`, and `pastBlocksPriorityPercentileBasisPoints` so fee estimation can be tuned separately for mempool estimator vs past blocks estimator
- compute fee bucket min for below congestion threshold case using `computeFeeBucketExponentFromFeeNanosPerKB` to avoid rounding error
- Use last fee time ordered transaction for computing fee time bucket range when priorityPercentileBasisPoints is 0.
- Update `sortCachedBlocks` to use block hash as a tie breaker.

### CODE REVIEW

**Comment by lazynina** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/deso-protocol/core/887?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `feature/proof-of-stake`
  * **PR #878** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/878?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #887** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/887?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/deso-protocol/core/887?utm_source=stack-comment).

File: lib/pos_fee_estimator.go [hunk 1]
```diff
@@ -404,19 +435,77 @@ func (posFeeEstimator *PoSFeeEstimator) estimateTxnFeeGivenTransactionRegister(
...
+// priorityPercentileBasisPoints = 2000 (20%)
+//
+// With these numbers, I think we'd have the following behavior (and notice I'm getting rid of the case where we keep
+// the fee the same, see comment below):
+// - We find the fee of the 20th percentile txn, which is enough to be included in the next block
+// - If the mempool has more than 5 blocks worth of txns, we pay one fee bucket higher than this guy, otherwise we pay
+// one fee bucket lower than him.
+//
+// This one is a bit weird, and I think the simple single-block thing previously would be better but lmk what you think
+// of it.
```
Comment (tholonious) [inline lib/pos_fee_estimator.go:472]:
```suggestion
```

File: lib/pos_fee_estimator.go [hunk 2]
```diff
@@ -442,41 +531,36 @@ func (posFeeEstimator *PoSFeeEstimator) estimateFeeRateNanosPerKBGivenTransactio
...
 	// but less than the max size of num blocks, we return the Priority fee.
-	if totalTxnsSize > congestionThreshold && totalTxnsSize <= maxSizeOfNumBlocks {
-		// Return Priority fee
-		return bucketMinFee, nil
-	}
+	//if totalTxnsSize > congestionThreshold && totalTxnsSize <= maxSizeOfNumBlocks {
+	//	// Return Priority fee
+	//	return bucketMinFee, nil
+	//}
 	// Otherwise, we return one bucket higher than Priority fee
```
Comment (tholonious) [inline lib/pos_fee_estimator.go:564]: Do we need this commented out block? If not, please delete

File: lib/pos_mempool.go [hunk 3]
```diff
@@ -190,6 +203,16 @@ func (mp *PosMempool) Start() error {
 		return errors.New("PosMempool.Start: PosMempool not initialized")
 	}
 
+	// Create the transaction register, the ledger, and the nonce tracker,
+	mp.txnRegister = NewTransactionRegister(mp.globalParams)
+	mp.feeEstimator = NewPoSFeeEstimator()
+	// TODO: parameterize num mempool blocks and num past blocks. Also, how to pass in blocks.
+	if err := mp.feeEstimator.Init(mp.txnRegister, 1, []*MsgDeSoBlock{}, 60, mp.globalParams); err != nil {
+		return errors.Wrapf(err, "PosMempool.Start: Problem initializing fee estimator")
+	}
```
Comment (tholonious) [inline lib/pos_mempool.go:212]: Any reason we call `mp.feeEstimator.Init(...)` here in the `PosMempool.Start(...)` if it's already called in `PosMempool.Init(...)` above? The feeEstimator should already be initialized with the right blocks before the `PosMempool.Start(...)` function is called
```suggestion
```

**Review by tholonious — APPROVED** [review]

**Pull request merged**